### PR TITLE
DIRECTOR: Bunch of stuff

### PR DESCRIPTION
**Pull request by einstein95**
There is a ton of stuff in here that spans from my last Director PR to now, including:
* try to get XFCNs started
* Fix finding files in punycoded foldernames
* Correctly decode non-ASCII alerts
* fileio: I have no idea if this fixes or breaks things
* testExtensions: Try the non-converted filename as well
* Add XFCNs for Flintstones Coloring Book
* Add full readme for PopUpMenuXObj
* hack: fix bug where _loopEnd can be 0 with _loopStart > 0
* Start adding XFCNs/Xlibs used by Ednovation games
* Add D4 hybrid release of Alice (serial: TORT-3001)
* Fix loading Xlibs from non-parent movies
* Add detection for amandastories v4.0; patches for Windows version
* Add IsPippin XCMD for Circus! (Pippin)
* Fix m_put not returning item
* Blender 1.5 demos and detection fixes
* Add Voodoo Lounge

### CODE REVIEW

File: engines/director/lingo/xlibs/darkenscreen.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/*************************************
+ *
+ * USED IN:
+ * Secrets of the Pyramids (Mac)
+ * Mindscape's Wonder Land (Mac)
+ * Grackon's Curse (Mac)
+ * Return to Jurassic (Mac)
+ * Ednovation demos (Mac)
+ *
+ *************************************/
```
Comment (sev-) [inline engines/director/lingo/xlibs/darkenscreen.cpp:31]: If it has `mDescribe`, please paste it here.

File: engines/director/lingo/xlibs/fileexists.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	}
+	Common::String filename = lastPathComponent(path, g_director->_dirSeparator);
+	if (!(saves->exists(filename))) {
+		Common::File *f = new Common::File;
+
+		if (!f->open(Common::Path(pathMakeRelative(origpath), g_director->_dirSeparator))) {
+			g_lingo->push(Datum(false));
+			return;
+		}
+	g_lingo->push(Datum(true));
```
Comment (sev-) [inline engines/director/lingo/xlibs/fileexists.cpp:78]: indentation is off

File: engines/director/lingo/xlibs/iscd.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*************************************
+ *
+ * USED IN:
+ * Secrets of the Pyramids (Mac)
+ *
+ *************************************/
```
Comment (sev-) [inline engines/director/lingo/xlibs/iscd.cpp:27]: no `mDescribe`?

File: engines/director/lingo/xlibs/ispippin.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*************************************
+ *
+ * USED IN:
+ * Circus! (Pippin)
+ *
+ *************************************/
```
Comment (sev-) [inline engines/director/lingo/xlibs/ispippin.cpp:27]: `mDescribe`?

File: engines/director/lingo/xlibs/moovxobj.h [hunk 5]
```diff
@@ -19,6 +19,7 @@
  *
  */
 
+
```
Comment (sev-) [inline engines/director/lingo/xlibs/moovxobj.h:22]: Very important change ;)

**Review by sev- — APPROVED** [review]
A good set of changes. Have only a few small notes.

File: engines/director/detection_tables.h [hunk 6]
```diff
@@ -1933,10 +1946,10 @@ static const DirectorGameDescription gameDescriptions[] = {
 	// English Mac filename is The C.H.A.O.S. Continuum™
 	// Japanese Mac demo filename is CHAOS デモ and another file is called CHAOS Demo
 	// English Mac demo from The Family Doctor, 3rd Edition
-	MACGAME1("chaos", "", "xn--The C.H.A.O.S. Continuum-3r4o", "2ce360c9ea2da80a2c6d1040b0ad92dd", 384225, 300),
+	MACGAME1("chaos", "", "xn--The C.H.A.O.S. Continuum-3r4o", "2ce360c9ea2da80a2c6d1040b0ad92dd", 384225, 313),
 	WINGAME2t("chaos", "", "CHAOS.EXE",    "5d1d9e1ff9a43e4d5b28441ce6d642fd", 375294,
 						   "ANTEROOM.MMM", "6004a72285c7757e12ea214374e6fc5d", 15160, 310),
-	MACDEMO1_l("chaos", "Demo", "Start CHAOS Demo", "f5277c53bacd27936158dd3867e587e2", 394683, Common::JA_JPN, 311),
+	MACDEMO1_l("chaos", "Demo", "Start CHAOS Demo", "f5277c53bacd27936158dd3867e587e2", 394939, Common::JA_JPN, 311),
```
Comment (tag2015) [inline engines/director/detection_tables.h:1952]: 394683 is the correct filesize

File: engines/director/detection_tables.h [hunk 7]
```diff
@@ -2415,12 +2436,20 @@ static const DirectorGameDescription gameDescriptions[] = {
 									"BASEB1.MMM",	"6cd090073aa3e159490d4a0458adaf7e", 1251222, 300),
 
 	MACGAME1("msdinosaurs", "Overview Movie", "Overview Movie", "479541070b3cbab6fd8473ce6267fb98", 792604, 310),
+	// From CD Loisirs Nº6
+	WINDEMO2t("msdinosaurs", "Demo", "DINOSAUR.EXE", "593f33ddf837f0b9c737712351ff6faa", 370013,
+									 "DINODEMO.MMM", "78863a9c384aea75affff8bb61294751", 2959664, 300),
+
+	// From CD Loisirs Nº6
+	WINDEMO1t("msdangerous", "", "DANGER.EXE", "04af79d00b4492d86baff5fe3601f534", 370004, 300),
+	WINDEMO1t("msspaceflight", "", "SPACE.EXE", "66eeaad4eb9d8ed1ad9c445ed4eacb14", 370004, 300),
```
Comment (tag2015) [inline engines/director/detection_tables.h:2445]: These two are very common, could you please add another file?

File: engines/director/detection_tables.h [hunk 8]
```diff
@@ -4305,10 +4448,19 @@ static const DirectorGameDescription gameDescriptions[] = {
 	WINGAME1("painters", "", "PAINTERS.EXE", "3cba94b585493ef5f22d0d650b6eb7fd", 895727, 404),
 	MACGAME1("painters", "", "Painters Painting", "0a81aee4106eec4ffc0cc564a7c15a23", 288774, 404),
 
+	WINDEMO1t("panicpark", "Demo", "PANIC.EXE", "9d0fe435d9becfcc67f7c25ad1f88102", 2370615, 404),
```
Comment (tag2015) [inline engines/director/detection_tables.h:4451]: Same as above

File: engines/director/detection_tables.h [hunk 9]
```diff
@@ -4339,6 +4491,18 @@ static const DirectorGameDescription gameDescriptions[] = {
...
+	MACGAME1_l("pepperon", "", "xn--u9j2la0f4a9404c9vg8wz", "8b138db44d4421cc7294a9dc792ccf1b", 521283, Common::JA_JPN, 404),
+	// Decomposed variant of the above
+	MACGAME1_l("pepperon", "", "xn--u9jtea0wb1ktbx480dksir53a", "8b138db44d4421cc7294a9dc792ccf1b", 521283, Common::JA_JPN, 404),
+	WINGAME1t_l("pepperon", "", "PEPPERON/START95.EXE", "b7fd4c43d44291a15ef1829e96652a5b", 1094585, Common::JA_JPN, 404),
+
+	// French version titled "Valmaison au fil des saisons"
+	MACGAME1_l("pepperon", "", "Valmaison", "8b138db44d4421cc7294a9dc792ccf1b", 502569, Common::FR_FRA, 404),
+	WINGAME1_l("pepperon", "", "PEPPERON/START95.EXE", "4086df04abc18956581cee17b48e81c1", 1805439, Common::FR_FRA, 404),
+	// Dutch version titled "Op avontuur in kabouterstad"
+	WINGAME1_l("pepperon", "", "START95.EXE", "e67abd6a890060fa8c41b7b363013254", 1088309, Common::NL_NLD, 404),
```
Comment (tag2015) [inline engines/director/detection_tables.h:4504]: If there's no subfolder this one too needs an additional file

File: engines/director/detection_tables.h [hunk 10]
```diff
@@ -5266,40 +5470,30 @@ static const DirectorGameDescription gameDescriptions[] = {
...
+
+	MACGAME2("explorespace4", "", "Galaxy",				 "231d3041df162be4a0ddd36a74eb03b6", 719005,
+								  "PROGRAM/460MAIN.Dxr", "4212a73dc3482104192ad4f0937446cf", 275180, 501),
+	WINGAME2("explorespace4", "", "WIN95.EXE",			 "1a7acbba10a7246ba58c1d53fc7203f5", 1411387,
+								  "PROGRAM/460MAIN.DXR", "4212a73dc3482104192ad4f0937446cf", 275180, 501),
+
+	MACGAME2("explorespace5", "", "Space",				 "231d3041df162be4a0ddd36a74eb03b6", 719005,
+								  "PROGRAM/459MAIN.Dxr", "3868e98be38c2fbff814e2b8f1663fc9", 376868, 501),
+	WINGAME2("explorespace5", "", "WIN95.EXE",			 "1a7acbba10a7246ba58c1d53fc7203f5", 1411387,
+								  "PROGRAM/459MAIN.DXR", "3868e98be38c2fbff814e2b8f1663fc9", 376868, 501),
```
Comment (tag2015) [inline engines/director/detection_tables.h:5497]: win 3,1 entries are unneeded?

File: engines/director/lingo/xlibs/darkenscreen.cpp (hunk 1 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/darkenscreen.cpp:31]: It doesn't have any `mDescribe` output

File: engines/director/lingo/xlibs/iscd.cpp (hunk 3 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/iscd.cpp:27]: It doesn't have any `mDescribe` output

File: engines/director/lingo/xlibs/ispippin.cpp (hunk 4 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/ispippin.cpp:27]: It doesn't have any `mDescribe` output

File: engines/director/detection_tables.h [hunk 11]
```diff
@@ -2324,9 +2342,9 @@ static const DirectorGameDescription gameDescriptions[] = {
 									"MSI C-S",   "224006f035e5f5fc0dd0133fbc5c6b89", 142729, Common::JA_JPN, 311),
 
 	// Original Mac filename is Maniac Sports™
-	MACGAME1("maniacsports", "", "xn--maniac sports-8q9g", "2ce360c9ea2da80a2c6d1040b0ad92dd", 420962, 311),
-	WINGAME2("maniacsports", "", "MANIAC.EXE", "65d06b5fef155a2473434571aff5bc29", 371274,
-								 "SUPPORT/MSWORLD.MMM", "e9cdd7f30fb3386e7b67c6769147f60a", 218378, 311),
+	MACGAME1("maniacsports", "", "xn--Maniac Sports-8q9g", "2ce360c9ea2da80a2c6d1040b0ad92dd", 420962, 311),
+	WINGAME2t("maniacsports", "", "MANIAC.EXE", "06c073b9e084e7f41c814f00229e3cde", 371274,
+								  "SUPPORT/MSINTRO.MMM", "02bfadbf255139d60a0b9dd1382936d1", 377340, 311),
```
[the remark this answers is not in the document]
Comment (einstein95) [reply engines/director/detection_tables.h:2347]: No particular reason, I just like to use the file that is launched by the EXE, either directly or via lingo script

File: engines/director/detection_tables.h (hunk 10 above)
Comment (einstein95) [reply engines/director/detection_tables.h:5497]: Yes, the general approach is to use, where available, Win9x binaries over Win3.x and PowerPC/fat binaries over 68k

File: engines/director/detection_tables.h (hunk 6 above)
Comment (einstein95) [reply engines/director/detection_tables.h:1952]: Fixed

File: engines/director/lingo/xlibs/moovxobj.h (hunk 5 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/moovxobj.h]: Fixed

File: engines/director/lingo/xlibs/fileexists.cpp (hunk 2 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/fileexists.cpp]: Fixed

File: engines/director/detection_tables.h (hunk 8 above)
Comment (einstein95) [reply engines/director/detection_tables.h]: Done

File: engines/director/detection_tables.h (hunk 9 above)
Comment (einstein95) [reply engines/director/detection_tables.h]: Done

File: engines/director/detection_tables.h [hunk 12]
```diff
@@ -4798,8 +4985,17 @@ static const DirectorGameDescription gameDescriptions[] = {
 	WINGAME1("vnc", "", "VNC.EXE", "40ba00213a10164eb6e01847108f9b21", 1086869, 404),
 	WINGAME1("vnc", "Beta", "VNC.EXE", "e6f284971c09f19e3277aa8ebcf58cbd", 726643, 404),
 
+	// Mac version requires installation, 'Install Voodoo Lounge', StuffIt
+	MACGAME1("voodoolounge", "", "Voodoo Lounge", "b7e69c37b7355022d400c14aa97c5d54", 502080, 404),
+	WINGAME1t("voodoolounge", "", "VOODOO.EXE", "09553ceec915a1eeb9a990245bfd189b", 866187, 404),
```
[the remark this answers is not in the document]
Comment (einstein95) [reply engines/director/detection_tables.h]: Done

**Comment by sev-** [comment]
Thank you!

**Pull request merged**